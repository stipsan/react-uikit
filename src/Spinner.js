import { PropTypes, PureComponent } from 'react'
import cx from 'classnames'
import computedStyle from 'computed-style'

class Spinner extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
  }

  static defaultProps = {
    className: '',
  }

  state = { size: 0, strokeWidth: 0 }

  componentDidMount() {
    this.setMeasurements()
  }

  componentWillUpdate() {
    this.setMeasurements()
  }

  setMeasurements = () => {
    this.setState({
      size: this.container.clientWidth,
      strokeWidth: parseFloat(computedStyle(this.circle, 'stroke-width')),
    })
  }

  setContainer = (node) => {
    if (node) {
      this.container = node
    }
  }

  setCircle = (node) => {
    if (node) {
      this.circle = node
    }
  }

  render() {
    const { className } = this.props
    const { size, strokeWidth } = this.state
    const viewBox = `0 0 ${size} ${size}`
    const diameter = Math.floor(size / 2)

    return (
      <span className={cx('uk-icon uk-spinner', className)} ref={this.setContainer}>
        <svg
          height={size}
          ratio="1"
          viewBox={viewBox}
          width={size}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx={diameter}
            cy={diameter}
            fill="none"
            r={diameter - strokeWidth}
            ref={this.setCircle}
            stroke="#000"
          />
        </svg>
      </span>
    )
  }
}

export default Spinner
