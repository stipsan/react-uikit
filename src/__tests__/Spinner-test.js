import renderer from 'react-test-renderer'
import Spinner from '../Spinner'

jest.mock('react-dom')

it('should render correctly', () => {
  const component = renderer.create(<Spinner />)
  console.log(component)
  expect(component.toJSON()).toMatchSnapshot()
})
