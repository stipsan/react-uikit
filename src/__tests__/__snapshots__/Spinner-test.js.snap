// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<span
  className="uk-icon uk-spinner"
>
  <svg
    height={32}
    ratio="1"
    viewBox="0 0 32 32"
    width={32}
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx={16}
      cy={16}
      fill="none"
      r={15}
      stroke="#000"
    />
  </svg>
</span>
`;
